src/app/publications/publications.component.ts

```diff
@@ -0,0 +1,89 @@
+import { ChangeDetectorRef, Component, OnInit } from '@angular/core';
+import { MatBottomSheet, MatDialog } from '@angular/material';
+import { PublicationService } from './publication.service';
+import { PublicationsNewComponent } from './publications-new.component';
+import { PublicationResponse } from './publication-response';
+import { catchError, filter } from 'rxjs/operators';
+import { HttpErrorResponse } from '@angular/common/http';
+import { Observable, of } from 'rxjs';
+import { GlobalErrorHandlerService } from '../error/global-error-handler.service';
+import { DeleteConfirmationDialogComponent } from '../shared/delete-confirmation-dialog/delete-confirmation-dialog.component';
+import { PublicationsEditComponent } from './publications-edit.component';
+
+@Component({
+  selector: 'app-publications',
+  templateUrl: './publications.component.html',
+  styleUrls: ['./publications.component.scss']
+})
+export class PublicationsComponent implements OnInit {
+
+  errorMessage: string = null;
+  publications$: Observable<PublicationResponse[]> = of([]);
+
+  constructor(private publicationService: PublicationService,
+              public dialog: MatDialog,
+              private changeDetector: ChangeDetectorRef,
+              private globalErrorHandlerService: GlobalErrorHandlerService,
+              private bottomSheet: MatBottomSheet) {
+  }
+
+  ngOnInit() {
+    this.loadPublications();
+  }
+
+  openNewDialog() {
+    this.bottomSheet.open(PublicationsNewComponent)
+      .afterDismissed().subscribe((publication: PublicationResponse) => {
+      if (publication) {
+        this.loadPublications();
+      }
+    });
+  }
+
+  openEditDialog(publication: PublicationResponse) {
+    this.bottomSheet.open(PublicationsEditComponent, {
+      data: <PublicationResponse>{
```

Comment:
I suggest to use `Object.assign({}, publication)` here. What do you think?

Reply:
thank you. done